### DISCUSSION TITLE
Floating number & integer input field

### DISCUSSION THREAD

**Discussion by lukaszmielczarekdev** [react-hook-form/react-hook-form#8662] [Q&A] [answered]
Hi,
I can't figure out how to make an input field which accepts floating numbers and integers. 
With this below I can only put integers, but I need to put something like 2.5 or 25.

`<input
                    type="number"
                    {...registerProduct("carbs", {
                      valueAsNumber: true,
                      max: 5000,
                      min: 0,
                      required: true,
                      maxLength: 4,
                      pattern: /\d+/,
                    })}
                  />`

I tried with different pattern, but it doesn't work.

Can you tell me what am I doing wrong?

**Comment by Moshyfawn** [comment, marked as answer]
You question is not related to the RHF implementation, but rather the basic HTML input `type="number"` API.

What you want is to add the `step` attribute to your `input` element
```js
<input step="0.01" />
```

**Reply by Moshyfawn** [reply]
Also, on the topic of `pattern` attribute usage with type="number" inputs, see the #8661 discussion

**Reply by lukaszmielczarekdev** [reply]
Thank you. Everything works fine now.
